src/main/java/org/broadinstitute/hellbender/tools/walkers/validation/ReferenceBlockConcordance.java

```diff
@@ -0,0 +1,215 @@
+            }
+        }
+
+        currentTruthVariantContext = null;
+        currentEvalVariantContext = null;
+        currentContig = null;
+    }
+
+    private void evaluateNewContig(TruthVersusEval truthVersusEval) {
+        // If not beginning of file
+        if (currentContig != null) {
+            evaluateEndOfContig();
+        }
+
+        currentContig = truthVersusEval.getTruthIfPresentElseEval().getContig();
+    }
+
+    @Override
+    protected void apply(TruthVersusEval truthVersusEval, ReadsContext readsContext, ReferenceContext refContext) {
+        // New contig or beginning of file
+        if (!truthVersusEval.getTruthIfPresentElseEval().getContig().equals(currentContig)) {
+            evaluateNewContig(truthVersusEval);
+        }
+
+        // Evaluate only when currently seeing two NON_REF blocks
+        if (currentTruthVariantContext != null && currentEvalVariantContext != null) {
+            int blockStart = Math.max(currentTruthVariantContext.getStart(), currentEvalVariantContext.getStart());
+            int blockEnd = Math.min(currentTruthVariantContext.getEnd(), currentEvalVariantContext.getEnd());
+            int jointBlockLength = blockEnd - blockStart + 1;
+            // It is possible that jointBlockLength is negative if there is a gap in one file and the start of a new block in the other file.
+            // Since there is no overlap though, we can just skip that case.
+            if (jointBlockLength > 0) {
+                confidenceConcordanceHistogram.increment(new Pair<>(currentTruthVariantContext.getGenotype(0).getGQ(), currentEvalVariantContext.getGenotype(0).getGQ()).toString(), blockEnd - blockStart + 1);
+            }
+
+            int currentPosition = truthVersusEval.getTruthIfPresentElseEval().getStart();
+            if (truthVersusEval.hasTruth() || currentPosition >= currentTruthVariantContext.getEnd()) {
+                currentTruthVariantContext = null;
+            }
+            if (truthVersusEval.hasEval() || currentPosition >= currentEvalVariantContext.getEnd()) {
+                currentEvalVariantContext = null;
+            }
+        }
+
+        // Truth
+        if (truthVersusEval.hasTruth() && isNonRef(truthVersusEval.getTruth())) {
+            currentTruthVariantContext = truthVersusEval.getTruth();
+
+            // The end is inclusive, thus the plus one when calculating the length
+            int blockLength = truthVersusEval.getTruth().getEnd() - truthVersusEval.getTruth().getStart() + 1;
+
+            // TODO can a non_ref block ever have a number of genotypes != 1?
+            if(truthVersusEval.getTruth().getGenotypes().size() != 1) {
+                throw new IllegalStateException(String.format("The NON_REF block \"%s\" has more than one genotype, which is not supported.", truthVersusEval.getTruth().toStringDecodeGenotypes()));
+            }
```

Comment:
In every case I have come across the number of genotypes is always 1 for a symbolic variant, otherwise it would be of type "mixed". Or is there a case I haven't considered?

Reply:
done